File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> set = new HashSet<>();

        for (int num : nums) {
            set.add(num);
        }

        int best = 0;
        for (int num : nums) {
            int curr = 1;
            int temp = num;
            if (set.contains(num - 1)) {
                continue;
            }
            while (set.contains(temp + 1)) {
                curr++;
                temp++;
            }

            best = Math.max(best, curr);

        }

        return best;
    }
}

File: Data Structures & Algorithms/jump-game-ii/submission-2.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        count = 0
        l, r = 0, 0

        while r < len(nums) - 1:
            furthest = 0
            for i in range(l, r + 1):
                furthest = max(furthest, i + nums[i])
            l = r
            r = furthest
            count += 1
        
        return count
